### ISSUE TITLE
Question about single GPU training

### ISSUE THREAD

**Issue by smg0411** [Gavinwxy/DGCL#12]
Thank you for your incredible work. I am currently planning to conduct experiments using the VOC 2012 dataset on my PC setup, which utilizes a single GeForce RTX 4090 GPU. I intend to modify the code to fit my environment for the experiment, and I'm curious if there are any performance differences when conducting the experiment on a single GPU. Additionally, I'd appreciate any advice on considerations or best practices for running experiments with a single GPU setup. Below is the code section I've modified; could you provide any comments on it
![image](https://github.com/Gavinwxy/DGCL/assets/106463978/824b8ecc-11fa-4a1f-9e2b-38af0e9b041e)
![image](https://github.com/Gavinwxy/DGCL/assets/106463978/6a706435-c026-4216-a7b7-d50379917fb7)
![image](https://github.com/Gavinwxy/DGCL/assets/106463978/f4df0070-34ae-4292-a858-6720622ec113)
![image](https://github.com/Gavinwxy/DGCL/assets/106463978/4e1ba03d-07ea-4a19-8b00-ccb086ca2759)
![image](https://github.com/Gavinwxy/DGCL/assets/106463978/ed2a7d81-6384-4ad8-af63-b8d40e6a7b15)
![image](https://github.com/Gavinwxy/DGCL/assets/106463978/1386d98d-10f0-4485-9a82-49c1264baa69)

**Comment by gavinwxy** [comment]
Since you have modified all the data collection and loss calculation from multiple to single gpu setting, the next thing you can try is to set the learning rate based on the rules [here](https://github.com/Lightning-AI/pytorch-lightning/discussions/3706#discussioncomment-238302). Personally, I have not try single GPU setting so you might need to carefully fine-tune the settings. However, since current SOTA methods all follow 4-GPU or 8-GPU setting, it is not guaranteed that you can achieve the same level of performance using single GPU.
